services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/asn1/SpatDecoder.java

```diff
@@ -0,0 +1,175 @@
+            try{
+                ProcessedSpat processedSpat = createProcessedSpat(spat);
+                return new SpatDecodedMessage(processedSpat, spat, message.getAsn1Message(), "");
+            } catch(Exception e) {
+                return new SpatDecodedMessage(null, spat, message.getAsn1Message(), e.getMessage());
+            }
+            
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+            return new SpatDecodedMessage(null, null, message.getAsn1Message(), e.getMessage());
+        } catch (Exception e) {
+            e.printStackTrace();
+            return new SpatDecodedMessage(null, null, message.getAsn1Message(), e.getMessage());
+        }
+    }
+
+    @Override
+    public OdeData getAsOdeData(String encodedData) {
+        OdeMsgPayload payload = new OdeAsn1Payload(new OdeHexByteArray(encodedData));
+
+        // construct metadata
+        OdeSpatMetadata metadata = new OdeSpatMetadata(payload);
+
+        metadata.setOdeReceivedAt(DecoderManager.getCurrentIsoTimestamp());
+        metadata.setOriginIp(DecoderManager.getStaticUserOriginIp());
+        metadata.setRecordType(RecordType.spatTx);
+        metadata.setSecurityResultCode(SecurityResultCode.success);
+
+        // construct metadata: receivedMessageDetails
+        ReceivedMessageDetails receivedMessageDetails = new ReceivedMessageDetails();
+        receivedMessageDetails.setRxSource(RxSource.NA);
+
+        // construct metadata: locationData
+        OdeLogMsgMetadataLocation locationData = new OdeLogMsgMetadataLocation();
+        receivedMessageDetails.setLocationData(locationData);
+
+        metadata.setReceivedMessageDetails(receivedMessageDetails);
+        metadata.setSpatSource(SpatSource.V2X);
+
+        Asn1Encoding unsecuredDataEncoding = new Asn1Encoding("unsecuredData", "MessageFrame",
+                EncodingRule.UPER);
+        metadata.addEncoding(unsecuredDataEncoding);
+
+        // construct odeData
+        return new OdeAsn1Data(metadata, payload);
+    }
+
+    @Override
+    public OdeSpatData getAsOdeJson(String consumedData) throws XmlUtilsException {
+        ObjectNode consumed = XmlUtils.toObjectNode(consumedData);
+
+		JsonNode metadataNode = consumed.findValue(AppContext.METADATA_STRING);
+		if (metadataNode instanceof ObjectNode) {
+			ObjectNode object = (ObjectNode) metadataNode;
+			object.remove(AppContext.ENCODINGS_STRING);
+			
+			//Spat header file does not have a location and use predefined set required RxSource
+			ReceivedMessageDetails receivedMessageDetails = new ReceivedMessageDetails();
+			receivedMessageDetails.setRxSource(RxSource.NA);
+			 ObjectMapper objectMapper = new ObjectMapper();
+			 JsonNode jsonNode;
+			try {
+				jsonNode = objectMapper.readTree(receivedMessageDetails.toJson());
+				object.set(AppContext.RECEIVEDMSGDETAILS_STRING, jsonNode);
+			} catch (JsonProcessingException e) {				
+				e.printStackTrace();
```

Comment:
chore(blocking): replace with proper logging
```suggestion
				log.error("Unable to convert receivedMessageDetails to JSON: {}", e.getMessage());
```